bftengine/src/bftengine/BFTEngine.cpp

```diff
@@ -190,6 +196,14 @@ IReplica::IReplicaPtr IReplica::createNewReplica(ReplicaConfig *replicaConfig,
   return replicaInternal;
 }
 
+IReplica::IReplicaPtr IReplica::createNewReplica(ReplicaConfig *replicaConfig,
+                                                 IRequestsHandler *requestsHandler,
+                                                 IStateTransfer *stateTransfer,
+                                                 bft::communication::ICommunication *communication,
+                                                 MetadataStorage *metadataStorage) {
+  bool dummy;
```

Comment:
What does this signify? As in prior comment it'd be better to pass an enum to the constructor instead of a bool.

Reply:
I didn't want to break the current API and implementation, so I have added another signature with the reference to bool in which we return if the metadata storage has been deleted on startup.